esss_fix_format/hooks.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import abc
+import textwrap
+
+
+def with_metaclass(meta, *bases):  # Borrowed from six.
+    """Create a base class with a metaclass."""
+
+    # This requires a bit of explanation: the basic idea is to make a dummy
+    # metaclass for one level of class instantiation that replaces itself with
+    # the actual metaclass.
+    class metaclass(type):
+
+        def __new__(cls, name, this_bases, d):
+            return meta(name, bases, d)
+
+        @classmethod
+        def __prepare__(cls, name, this_bases):
+            return meta.__prepare__(name, bases)
+
+    return type.__new__(metaclass, str('temporary_class'), (), {})
+
+
+class GitHook(with_metaclass(abc.ABCMeta, object)):
```

Comment:
Why do we need a metaclass here?

Reply:
I have no idea -- just moved the code from eden ;)

Reply:
hehehe.
OK.

I guess the original author was using if for [abc.abstractmethod](https://docs.python.org/2/library/abc.html#abc.abstractmethod).